### DISCUSSION TITLE
Select inside/around indentation level

### DISCUSSION THREAD

**Discussion by MyApaulogies** [helix-editor/helix#10796] [Ideas] [answered]
What if there was an action like `mii` that makes selections based on indentation levels? Here are some examples (left is before, right is after):

![image](https://github.com/helix-editor/helix/assets/49665239/08a00c85-2ed9-4f5b-8921-a008edf0c6c4)
![img2](https://github.com/helix-editor/helix/assets/49665239/dcc3ee43-0b6b-49ca-8ad7-10ed184b61f0)
![img3](https://github.com/helix-editor/helix/assets/49665239/c2a8e820-71eb-4d61-aded-693c059f486a)

The use case that made me think of this is traversing files full of bullet points similar to the pictures, and wanting to select all children bullet points to extract them, outdent, etc. 

Another use case would be selecting the contents of a `{}` pair to comment out, since `mi{` also selects the newline after the first `{`... I know that's a bit of a dumb example, since you can `A-;v` and adjust the border, but the point is that indentation can be more intuitive and useful to work with than just brackets or treesitter in some cases.

Lastly, this would be a useful tool in situations where you can't rely on treesitter, either when it breaks, the language you're working with lacks support, or your language doesn't have simple `{}` or `()` delimiters for easy `mi{` access (like XML).

There are a couple of details that might be wrong... The behavior of `mii` vs. `mai` in the pictures I just chose to match `mip` and `map`, but maybe it would make more sense for `mai` to extend *upwards* into blank lines (with Python/Yaml-like files in mind). Also maybe `mii` should be `min` or `mid` instead?

But that's the gist of it. Thanks for reading, I'd be curious to hear your thoughts.

**Comment by sibouras** [comment, marked as answer]
would love to have it too, there's a pr for it https://github.com/helix-editor/helix/pull/9843

**Reply by MyApaulogies** [reply]
ohh okay, awesome. thanks, I missed it

**Reply by TornaxO7** [reply]
please don't forget to mark your question as answered @MyApaulogies ;)

**Reply by MyApaulogies** [reply]
got it!

**Reply by TornaxO7** [reply]
![peepoAwesome](https://github.com/helix-editor/helix/assets/50843046/65de8b03-28cd-4a75-b098-0199405a1e44)
